code/code/misc/monster.h

```diff
@@ -115,7 +115,8 @@ class TMonster : public TBeing {
     // VIRTUAL FUNCTIONS
     virtual double baseDamage() const;
     virtual int grenadeHit(TTrap *);
-    virtual int checkSpec(TBeing *, cmdTypeT, const char *, TThing *);
+    using TThing::checkSpec;
```

Comment:
Which problem is this line fixing?

Reply:
Without that line, or if I had two separate overloads there with simply a different parameter type as the final parameter (one being the original TThing* and one being TObj*) I got tons of compiler warnings about hiding an overload whenever thing.h was used in the same file as monster.h (IIRC). From researching it, adding that using statement was one thing that was suggested on places like StackOverflow as a way to avoid those types of warnings.

Reply:
Hmmmm, how's the spec runner going to know which overload to call, anyway?